src/ssa/local_ssa.h

```diff
@@ -32,11 +34,14 @@ class local_SSAt
   inline local_SSAt(
     const goto_functiont &_goto_function,
     const namespacet &_ns,
+    const ssa_heap_analysist &_heap_analysis,
```

Comment:
I strongly advise not to pass this to the constructor, but create an instance during construction and pass it to the members.